ols/app/models/config.py

```diff
@@ -2,12 +2,25 @@
 
 import logging
 from typing import Dict, Optional
+from urllib.parse import urlparse
 
 from pydantic import BaseModel
 
 from ols import constants
 
 
+def is_valid_http_url(url) -> bool:
+    """Check is a string is a well-formed http or https URL."""
```

Comment:
let's use a regex for this so we can fully validate the url.  See: https://www.makeuseof.com/regular-expressions-validate-url/ for an example.

Reply:
Assuming OLS doesn't interpret provider and model URLs but simply passes them through, I don't think we should validate them too strictly ourselves or may be even not at all. I'd rather pass them along and let the code that makes use of them fail if it needs to, and we will report that error back to the user.
The example that you found thinks that "https://127.0.0.1:1234/" or "https://localhost:80/" are invalid URLs and we haven't touched IPv6 yet. ChatGPT generated me 
```
import re

def is_valid_url(url):
    regex = re.compile(
        r'^(https?://)'  # http:// or https://
        r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+[A-Z]{2,6}\.?|'  # domain
        r'localhost|'  # localhost
        r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}|'  # ...or ipv4
        r'\[?[A-F0-9]*:[A-F0-9:]+\]?)'  # ...or ipv6
        r'(?::\d+)?'  # optional port
        r'(?:/?|[/?]\S+)$', re.IGNORECASE)

    return re.match(regex, url) is not None
```
which at least attempts to handle IPv4 and IPv6 addresses. But unless there's a good reason for us to do the validation and we can lift the validation code from somewhere and that somewhere stays current on the RFCs that define URLs, I'd rather stay away from it. Otherwise we're going to hear users complaining that OLS is rejecting their valid URLs.

Reply:
I agree we don't want to define/implement/maintain our own validation.  I'd be very surprised if there's not an existing python lib that will do the validation for us so my proposal is to find+use one of those.  But if there's not clearly an existing library we can use for this, i'm fine w/ deferring it for now.

Reply:
But that is exactly what the code currently is doing, it's calling into just such a library https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse and checks that the scheme is http[s] and there's a netloc, so there's a network address to try to connect to. If you look at https://docs.python.org/3/library/urllib.parse.html#url-parsing-security, they're even admitting that "What constitutes a URL is not universally well defined." And if someone plugs in a bad URL, the network layer will figure it out. I just don't think this is something we need to get too deep into.

Reply:
ugh. i am very sorry.  My eyes went right over the "urlparse" call and directly into the checking for an http or https protocol.  this is fine.